FunctionalTableData/TableSection.swift

```diff
@@ -164,6 +164,20 @@ extension Array where Element: TableSectionType {
 
 extension TableSection: Equatable {
 	public static func ==(lhs: TableSection, rhs: TableSection) -> Bool {
-		return lhs.key == rhs.key
+		return lhs.key == rhs.key &&
+			(lhs.header == nil && rhs.header == nil || lhs.header?.isEqual(rhs.header) ?? false) &&
+			(lhs.footer == nil && rhs.footer == nil || lhs.footer?.isEqual(rhs.footer) ?? false) &&
+			isEqual(lhs: lhs.rows, rhs: rhs.rows) &&
+			lhs.style == rhs.style &&
+			(lhs.headerVisibilityAction == nil && rhs.headerVisibilityAction == nil || lhs.headerVisibilityAction != nil && rhs.headerVisibilityAction != nil) &&
+			(lhs.didMoveRow == nil && rhs.didMoveRow == nil || lhs.didMoveRow != nil && rhs.didMoveRow != nil)
```

Comment:
You shouldn't include the closures in the equality checks

Reply:
@g-Off said to include them like this, but I can remove them.

Reply:
A nil `didMoveRow` to non-nil, or vice-versa, should could as not being equal (same for the `headerVisibilityAction`)

Reply:
Added back.